common/src/View/EntityAttributeGridTable.cpp

```diff
@@ -412,21 +409,27 @@ namespace TrenchBroom {
             const StringList names = m_rows.names(pos, numRows);
             ensure(names.size() == numRows, "invalid number of row names");
             
-            const SetBool ignoreUpdates(m_ignoreUpdates);
-            
-            Transaction transaction(document, StringUtils::safePlural(numRows, "Remove Attribute", "Remove Attributes"));
+            {
+                const SetBool ignoreUpdates(m_ignoreUpdates);
```

Comment:
Note: I didn't modify this code, just indented and wrapped it in `{}` to limit the lifetime of the `SetBool`.